### PR TITLE
Switch Vision to use GAPIC/gRPC/GAX

### PR DESCRIPTION
**Pull request by blowmage**
Move Vision from the JSON API to the gRPC API, using GAPIC as provided by GAX.

### CODE REVIEW

File: google-cloud-vision/lib/google/cloud/vision/service.rb [hunk 1]
```diff
@@ -66,9 +77,13 @@ def inspect
         protected
 
         def execute
+          require "grpc" # Ensure GRPC is loaded before rescuing exception
           yield
-        rescue Google::Apis::Error => e
+        rescue GRPC::BadStatus => e
```
Comment (quartzmo) [inline google-cloud-vision/lib/google/cloud/vision/service.rb]: Is this `GRPC::BadStatus` needed? I didn't add it in conversions for Datastore and Pub/Sub. I can, however.

File: google-cloud-vision/lib/google/cloud/vision/v1/image_annotator_api.rb [hunk 2]
```diff
@@ -46,7 +46,7 @@ class ImageAnnotatorApi
           # The default port of the service.
           DEFAULT_SERVICE_PORT = 443
 
-          CODE_GEN_NAME_VERSION = "gapic/0.1.0".freeze
+          CODE_GEN_NAME_VERSION = "grpcc/0.1.0".freeze
```
Comment (quartzmo) [inline google-cloud-vision/lib/google/cloud/vision/v1/image_annotator_api.rb]: Looks like this was mistakenly changed in a `/s gapi grpc` search and replace!

**Comment by quartzmo** [comment]
In `vision.rb`, there is a docs section "Configuring retries and timeout". At the least, the `retries` mentions need to be removed. If you're feeling brave, you can add docs for `client_config`.

In my upcoming Pub/Sub conversion PR, this is what I have:

```
    # ## Configuring timeout
    #
    # You can configure the request `timeout` value in seconds.
```

File: google-cloud-vision/lib/google/cloud/vision/service.rb (hunk 1 above)
Comment (blowmage) [inline google-cloud-vision/lib/google/cloud/vision/service.rb]: I was trying to be cautious, but not rescuing `GRPC::BadStatus` would actually simplify things a bit. Updated.

File: google-cloud-vision/lib/google/cloud/vision/v1/image_annotator_api.rb (hunk 2 above)
Comment (blowmage) [inline google-cloud-vision/lib/google/cloud/vision/v1/image_annotator_api.rb]: Dang it, I checked for this many, many times. Can't believe this one got through. Updated.

**Comment by blowmage** [comment]
Great catches. I've removed the `retries` documentation.

**Comment by quartzmo** [comment]
This is so awesome! Thanks for figuring out the little details with gRPC requires in `Service`, I will add those to #903 and #959. Merging!

**Pull request merged**